svtools/lmerge.py

```diff
@@ -530,80 +504,82 @@ def r_cluster(BP_l, sample_order, v_id, use_product, include_genotypes=False):
+            v_id = merge(BP_r, sample_order, v_id, use_product, vcf, vcf_out, include_genotypes, weight_by_support, weight_by_samples)
             BP_r = [b]
             BP_max_end_r = b.end_r
             BP_chr_r = b.chr_r
 
     if len(BP_r) > 0:
-        v_id = merge(BP_r, sample_order, v_id, use_product, include_genotypes)
+        v_id = merge(BP_r, sample_order, v_id, use_product, vcf, vcf_out, include_genotypes, weight_by_support, weight_by_samples)
 
     return v_id
 
-def l_cluster_by_line(file_name, percent_slop=0, fixed_slop=0, use_product=False, include_genotypes=False):
-    v_id = 0
-    vcf_lines = []
-    vcf_headers = list()
-    infile=open(file_name, 'r')
 
-    header = ''
-    samples = ''
 
-    for l in infile:
-      if l[0] == '#':
-        if l[1] != '#':
-          samples = l.rstrip().split('\t')[9:]
-        else:
-          # ignore fileDate
-          if l[:10] == '##fileDate':
-            continue
-          if l not in vcf_headers:
-            vcf_headers.append(l)
-      if l[0] != '#':
-        break
-
-    sample_order = []
-    for header in vcf_headers:
-      if header[:8] == '##SAMPLE':
-        sample_order.append(header.rstrip()[13:-1])
-      print header,
-
-    # Add in the sample names etc
-    chromosome_header_line = '#CHROM\tPOS\tID\tREF\tALT\tQUAL\tFILTER\tINFO'
-    if include_genotypes:
-        sample_header = '\t'.join(sample_order)
-        chromosome_header_line += '\tFORMAT\t' + sample_header
-    print chromosome_header_line
-
-    BP_l = []
-    BP_sv_type = ''
-    BP_max_end_l = -1
-    BP_chr_l = ''
-
-    b = Breakpoint(l_bp.parse_vcf_record(l), percent_slop=percent_slop, fixed_slop=fixed_slop)
-    BP_l.append(b)
-    BP_max_end_l = max(BP_max_end_l, b.end_l)
-    BP_chr_l = b.chr_l
-    BP_sv_type = b.sv_type
-
-    for l in infile:
-      b = Breakpoint(l_bp.parse_vcf_record(l), percent_slop=percent_slop, fixed_slop=fixed_slop)
-      if (len(BP_l) == 0) or ((b.start_l <= BP_max_end_l) and (b.chr_l == BP_chr_l) and (b.sv_type == BP_sv_type)):
-        BP_l.append(b)
-        BP_max_end_l = max(BP_max_end_l, b.end_l)
-        BP_chr_l = b.chr_l
-        BP_sv_type = b.sv_type
-      else:
-        v_id = r_cluster(BP_l, sample_order, v_id, use_product, include_genotypes)
-        BP_l = [b]
-        BP_max_end_l = b.end_l
-        BP_sv_type = b.sv_type
-        BP_chr_l = b.chr_l
-
-    if len(BP_l) > 0:
-        v_id = r_cluster(BP_l, sample_order, v_id, use_product, include_genotypes)
-
-    infile.close()
+def l_cluster_by_line(file_name, percent_slop=0, fixed_slop=0, use_product=False, include_genotypes=False, weight_by_support=False, weight_by_samples=False):
+    
+    v_id = 0
+
+    in_header = True
+    header = []
+    vcf = Vcf()
+    vcf_out=sys.stdout
+    if weight_by_samples and weight_by_support:
+        sys.stderr('Error:  Can only weight by total support OR carrier count, but not both')
+        sys.exit(1)
```

Comment:
This check can be removed if the options for these are changed to be 'choice' style. If we weren't to do this, I'd advocate for placing this check directly after command line parsing.